vllm/entrypoints/llm.py

```diff
@@ -1037,7 +1038,8 @@ def _validate_and_add_requests(
         for i, prompt in enumerate(prompts):
             self._add_request(
                 prompt,
-                params[i] if isinstance(params, Sequence) else params,
+                params[i]
+                if isinstance(params, Sequence) else copy.copy(params),
```

Comment:
This copy is likely going to lower the throughput, any chance that we can remove it?

Reply:
Okay, I can move this into the function where we get the guided decoding logit processor